src/models.ts

```diff
@@ -297,3 +304,33 @@ export class AdvancedFilter extends Filter {
     return filter;
   }
 }
+
+export interface IDataReference {
+  target: IFilterTarget;
+}
+
+export interface IEqualsDataReference extends IDataReference {
+  equals: string | boolean | number;
+}
+
+export interface IBetweenDataReference extends IDataReference {
+  between: (string | boolean | number)[];
+}
+
+export interface IValueDataReference extends IDataReference {
+  value: string | boolean | number;
+}
+
+export interface IIdentityValue<T> {
```

Comment:
Hmm.... i'll try a combination of the two suggestions.  let me know what you think of next commit.